fw/apm.c

```diff
@@ -995,23 +996,6 @@ tfw_apm_stats_global(TfwPrcntlStats *pstats)
 				    read_unlock_bh);
 }
 
-/*
- * Verify that an APM Stats user using the same set of percentiles.
- *
- * Note: This module uses a single set of percentiles for all servers.
- * All APM Stats users must use the same set of percentiles.
- */
-int
-tfw_apm_pstats_verify(TfwPrcntlStats *pstats)
```

Comment:
Is there any particular reason why this function is removed? Just wondering. Thanks!

Reply:
I don't understand why it's required (and no one on the current team). In the single place where it was used (in `tfw_procfs_cfgend`) it checked that `tfw_pstats_ith` equals `tfw_pstats_ith`, that always be true. It seems that `tfw_pstats_ith` is hardcoded in several places as the only valid index array (`TfwPrcntlStats.ith`), but it was supposed flexibility that wasn't fully implemented.

Reply:
Or probably it's misunderstanding of the design, what was initial idea behind that?

Reply:
Yeah, that's what I thought. I don't fully recall what was the intended flexibility after all these years. As you mentioned, the percentiles were defined (hardcoded?) in several places in the code.

As far as I recall, this was the function to safeguard against changing the percentile arrangement in one place and not changing in another. Different people work on the code so it might be easy to miss or forget. As this function is only called once at startup, it's a cheap test that may prevent a lot of aggravation. So if that's no longer true and the percentile arrangement is defined in a single place as it should, then this function is no longer useful.

Reply:
Perhaps, then, this should go in the commit's comment.